tools/vscode/src/lib.rs

```diff
@@ -0,0 +1,538 @@
+pub struct LaunchConfig {
+    pub name: String,
+    pub r#type: String,
+    pub request: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub program: Option<String>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub args: Vec<String>,
+    pub cwd: String,
+    #[serde(rename = "sourceLanguages")]
+    pub source_languages: Vec<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub env: Option<BTreeMap<String, String>>,
+    #[serde(
+        rename = "targetCreateCommands",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub target_create_commands: Option<Vec<String>>,
+}
+
+/// VSCode task configuration for building.
+#[derive(Debug, Serialize)]
+pub struct TaskConfig {
+    pub label: String,
+    pub r#type: String,
+    pub command: String,
+    pub args: Vec<String>,
+    pub group: String,
+    pub presentation: TaskPresentation,
+    #[serde(rename = "problemMatcher")]
+    pub problem_matcher: Vec<String>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct TaskPresentation {
+    pub reveal: String,
+    pub panel: String,
+    #[serde(rename = "showReuseMessage")]
+    pub show_reuse_message: bool,
+    pub clear: bool,
+}
+
+/// Generator for VSCode launch configurations.
+pub struct LaunchConfigGenerator {
+    workspace_root: Utf8PathBuf,
+    bazel_binary: Utf8PathBuf,
+    bazel_info: BazelInfo,
+}
+
+impl LaunchConfigGenerator {
+    pub fn new(workspace_root: Utf8PathBuf, bazel_info: BazelInfo) -> Self {
+        Self {
+            workspace_root,
+            bazel_binary: "bazel".into(),
+            bazel_info,
+        }
+    }
+
+    pub fn with_bazel_binary(mut self, bazel: Utf8PathBuf) -> Self {
+        self.bazel_binary = bazel;
+        self
+    }
+
+    /// Query information about multiple targets at once.
+    pub fn query_targets_batch(&mut self, targets: &[String]) -> Result<Vec<TargetInfo>> {
+        if targets.is_empty() {
+            return Ok(vec![]);
+        }
+
+        // Get all target kinds in one query
+        let target_kinds = self.batch_query_target_kinds(targets)?;
+
+        let mut results = Vec::new();
+        for target in targets {
+            if let Some(target_kind) = target_kinds.get(target) {
+                let is_test = target_kind.contains("rust_test");
+
+                // We don't need to resolve binary paths now - that will happen at debug time
+                // Just create a placeholder path that will be resolved by the pre-launch task
+                results.push(TargetInfo {
+                    label: target.to_string(),
+                    binary_path: Utf8PathBuf::from(""), // Placeholder - resolved at debug time
+                    is_test,
+                    target_kind: target_kind.clone(),
+                });
+            }
+        }
+
+        Ok(results)
+    }
+
+    /// Query information about a specific target.
+    pub fn query_target_info(&mut self, target: &str) -> Result<TargetInfo> {
```

Comment:
AFAICT this is unused?

I was going to ask why we need both this and `find_rust_targets`, but I suspect the answer is we don't?